ledger/src/blockstore.rs

```diff
@@ -1030,7 +1017,10 @@ impl Blockstore {
 
         // Commit step: commit all changes to the mutable structures at once, or none at all.
         // We don't want only a subset of these changes going through.
-        write_batch.put_bytes::<cf::ShredCode>((slot, shred_index), &shred.payload)?;
+        let path = self.coding_shred_path(slot, shred_index);
+        let shred_path = Path::new(&path);
+        let mut f = fs::File::create(shred_path)?;
+        f.write_all(&shred.payload)?;
```

Comment:
Should we do the atomic update?, write to temp first, then do rename and also combine with the data logic

Reply:
Why?  I would rather get rid of the write lock on insert. Reconcile on read

Reply:
if you crash in the middle of writing a shred you don't end up with half a shred

Reply:
why do you need a lock on write with that?

Reply:
ah got it.  rename should be fast

Reply:
something like?
```
File::write_all()?;
atomic {
   write slot meta
   fs::rename(tmp_path, real_path)?;
}
```

then scanning the directory wont pick up any partially written

Reply:
@carllin seems like we only need the 'received' bit updated in SlotMeta on insert.  I can remove it and make received be a the highest index in the dir

Reply:
yea, that should be fast and hopefully more robust.

Reply:
@aeyakovenko hmmm, what do you mean only "need" the `received` bits?

If you look in `update_slot_meta`, it updates all the necessary fields at the end of an insert. These include:

`consumed`, `last_index` (updated when the is_last shred is detected), `received`, `completed_data_indexes` (array of indexes showing where each deserializable block of shreds is located). All of these could be potentially out of sync with the underlying data if not updated properly

Reply:
ok, got it.  how about the reconcile method?  i think then all we need is
1. write/rename to ensure full shreds
2. check if there are untracked shreds in get_slot_entries_with_shred_info